cmd/snap/cmd_run.go

```diff
@@ -112,6 +112,18 @@ and environment.
 		}, nil)
 }
 
+// isStopping returns true if the system is shutting down.
+func isStopping() (bool, error) {
+	cmd := exec.Command("systemctl", "is-system-running")
+	stdout, err := cmd.Output()
+	// systemctl is-system-running returns non-zero for comes other than "running"
+	// As such, ignore any ExitError and just process the stdout buffer.
+	if _, ok := err.(*exec.ExitError); ok {
+		return string(stdout) == "stopping\n", nil
```

Comment:
Wondering whether we should distinguish problems talking with systemd (probably reported to stderr?), from the result  which is written to stdout.

Reply:
We don't do anything about them anyway. We could maybe look at the fact that systemctl died on a signal but IMO YAGNI.

Reply:
I was thinking about this some more. On 14.04 there's no `is-system-running` but the logic handles that correctly today as we are looking precisely for the string `"stopping\n"`. Can you expand on ideas where we might want to handle the other values?

Reply:
I'm not sure. I brought it up for people to think about. I guess I'll do that now :)

`offline` says: "The manager is not running. Specifically, this is the operational state if an incompatible program is running as system manager (PID 1).". I guess this would be systems like trusty (ie, that are running enough of systemd on the side to support snaps) but that have `is-system-running`. I don't know if these exist so perhaps it is best to add a comment that if we find systems where `offline` is being returned, maybe we want to interrogate the system in another manner to see if it is in shutdown. The current logic of returning false seems like the best option for this hypothetical case for now.

`unknown` says "The operational state could not be determined, due to lack of resources or another error cause." Returning false for this seems reasonable enough, but *perhaps* we would want to retry in a way that won't overwhelm the system (eg, 5 times, each time after a second?). This is probably not required, though one could read the systemd source to determine what "or another error cause" actually means. On the other hand, if the operational state cannot be determined, maybe snapd should treat that the same as an error.